### ISSUE TITLE
formula_parser.py - errors

### ISSUE THREAD

**Issue by epogrebnyak** [epogrebnyak/make-xls-model#8]
1. After 18b8da220b7d48892c2a9bad09ad35111bd567e7 formula_parser.py fails newly added test. This happens because in expression like 'GDP[t] \* 0.5 + GDP[t-1] \* 0.5' variable name GDP appears twice and when using a dictionary to store variable name only one entry is stored. 

I beg to implement parsing in two stages: a) parse indices in brackets - the resulting string will include <varname>[<int>] , b) catch all (<varname>)[(<int>)]  and substitute with Excel formulas. 

This will also help make the code maintainable and readable, regex's are hard to follow.

2. Also in 18b8da220b7d48892c2a9bad09ad35111bd567e7 changed a regex shorthand function, becasue it was failing a test with similar variable names like "GDP" and "GDP_IQ". Todo: add shorthand call in main function. 

3. Function blocks
3.1.

``` python
     # Calculate row, column of excel cell
        try:
            var_offset = variables_dict[var] # Variable offset in file
        except KeyError:
            raise ValueError('Variable %s included in formula should be included in variables_dict' % repr(var))
```

please make it `var_offset = get_cell_row(...)`

3.2.

``` python
        period_normalize = re.sub('[' + ''.join(TIME_INDEX_VARIABLES) + ']', 't', period)
        t = time_period                            # Offset for reference period
        try:
            # We transfrom TIME_INDEX_VARIABLES to t for proper evaluation
            period_offset = eval(period_normalize)     # evaluate time expression t-1
        except:
            raise ValueError('Time expression %s[%s] invalid' % (var, period))
```

please make it `period_offset = get_cell_col(...)`

4. Why adding a print warning for a case when variable is in dictionary, but not in formula? This is a very common situation (formulas do not contain all variables) and it clogs the output. 

```
    if len(all_vars) != 0:
        for var in all_vars:
            if var != '':              
               # print('WARNING: Variable %s included in variables_dict but not present in formula' % repr(var))
```

**Comment by gabrielelanaro** [comment]
Ok, I'll complete the implementation and apply your changes as well. 

Should be able to do that later today (at least partially..)

**Comment by gabrielelanaro** [comment]
I started applying some cosmetic changes to eqcell_core.py, you can check the direction it is going, but I don't know if that's what you want.
Also, eqcell_core.py fails to run on my computer since commit 5a2c55e0c468502baca3b9986fd908ab353e94f4

**Comment by epogrebnyak** [comment]
I changed some code flow in parser - it now splits string like `GDP[1]+GDP[0]+0.5*GDP_IP[1]` to segments GDP[1],  GDP[0], GDP_IP[1] and substitutes them in the string with A1 refereneces. I like it better this way because change is more localised and there is no extra list of (varname, int) tuples.   

I think this closes this issue, let's discuss  eqcell_core.py in https://github.com/epogrebnyak/make-xls-model/issues/5
